src/acl-check.js

```diff
@@ -44,15 +44,45 @@ function accessDenied (kb, doc, directory, aclDoc, agent, modesRequired, origin,
   return ok
 }
 
+async function getTrustedModesForOrigin (kb, agent, origin) {
+  if (!kb || !agent || !origin) {
+    return Promise.resolve([])
+  }
+  const result = await query(`
+  SELECT ?mode WHERE {
+    ${agent} ${ACL('trustedApp')} ?trustedOrigin.
+    ?trustedOrigin  ${ACL('origin')} ${origin};
```

Comment:
The variable `?trustedOrigin` is a misnomer here, it refers to the *app* whose origin is `${origin}`

Reply:
Yes, we can change that.